pkg/kp/dsstore/dsstoretest/fake_dsstore.go

```diff
@@ -78,13 +109,67 @@ func (s *FakeDSStore) MutateDS(
 	id fields.ID,
 	mutator func(fields.DaemonSet) (fields.DaemonSet, error),
 ) (fields.DaemonSet, error) {
-	if _, ok := s.daemonSets[id]; ok {
-		newDS, err := mutator(s.daemonSets[id])
-		if err != nil {
-			return fields.DaemonSet{}, err
+	ds, _, err := s.Get(id)
+	if err != nil {
+		return fields.DaemonSet{}, err
+	}
+
+	ds, err = mutator(ds)
+	if err != nil {
+		return fields.DaemonSet{}, err
+	}
+
+	s.daemonSets[id] = ds
+
+	// TODO: If no one reads in these updates sent to this channel, it will block
+	if watcher, ok := s.watchers[id]; ok {
+		watched := FakeWatchedDaemonSet{
+			DaemonSet: &ds,
+			Operation: updated,
+			Err:       nil,
 		}
-		s.daemonSets[id] = newDS
-		return newDS, nil
+		watcher <- watched
 	}
-	return fields.DaemonSet{}, dsstore.NoDaemonSet
+
+	return ds, nil
+}
+
+func (s *FakeDSStore) Watch(quitCh <-chan struct{}) <-chan dsstore.WatchedDaemonSets {
+	outCh := make(chan dsstore.WatchedDaemonSets)
+
+	go func() {
+		for {
+			select {
+			case <-quitCh:
+				return
+			default:
+			}
+
+			outgoingChanges := dsstore.WatchedDaemonSets{}
+			// Reads in new changes that are sent to FakeDSStore.watchers
+			for _, ch := range s.watchers {
```

Comment:
No, nothing is added to s.watchers in this function but things do get added in the create, update, and delete.